### ISSUE TITLE
Add sympy/__main__.py

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#17181]
It should call isympy. This will make `python -m sympy` work. 

The one problem is that currently isympy has to not import sympy so that it can set some environment variables first (like for the cache or the ground types flags). I don't know if this is possible with `__main__.py`. So we should either figure out a way to do that, or else make it so those options can be changed after sympy has been imported.

**Comment by oscarbenjamin** [comment]
Is this related to pyoxidiser?

**Comment by asmeurer** [comment]
Not necessarily. It's good for modules that have command line tools to define `__main__.py`. That way you can use `python -m` and you don't have to worry about issues with potentially grabbing the wrong Python from the shebang line of a script. 

So for instance `isympy` uses `#!/usr/bin/env python`. If the first `python` on your `PATH` isn't the one you expected for whatever reason, it would launch the wrong thing. Or even possibly `isympy` itself could be coming from your `PATH`. A cleaner way is to pass the full path to `python` and use `-m isympy`. I am suggesting that `python -m sympy` should work in addition to `python -m isympy`, which currently works. But we should only do it if we can make it work correctly, either by making it not import sympy until the command line arguments are processed, or fixing those things so that it doesn't matter. So this is probably not an easy to fix issue.
